app/queries/comments/community_wellness_query.rb

```diff
@@ -0,0 +1,70 @@
+# Community Wellness Badge. These limitations consist of users that have at
+# least 2 comments without negative reactions (thumbsdown/vomit) per week.
+#
+# Note: Using `connection.execute("").to_a` instead of `.find_by_sql` because
+# ActiveRecord doesn't populate any other fields that aren't included in the raw
+# result set, so we don't get any added value from instantiating AR objects
+module Comments
+  class CommunityWellnessQuery
+    def self.call
+      ActiveRecord::Base.connection.execute(sql_query).to_a
+    end
+
+    def self.sql_query
+      # Based on the following Blazer query:
+      # https://dev.to/admin/blazer/queries/677
+      <<~SQL
+        SELECT user_id,
+               /* A comma separated string of "weeks_ago" */
+               array_to_string(array_agg(weeks_ago), ',') AS serialized_weeks_ago,
+               /* A comma separated string of comment counts. The first value in this string happens on the week that is the first value in serialized_weeks_ago */
+               array_to_string(array_agg(number_of_comments_with_positive_reaction), ',') AS serialized_comment_counts
+        FROM
+            (
+                SELECT user_id,
+                       COUNT(user_id) AS number_of_comments_with_positive_reaction,
+                       /* Get the number of weeks, since today for posts */
+                       (trunc((extract(epoch FROM (current_timestamp- created_at))) / 604800) + 1) AS weeks_ago
+                FROM comments
+                INNER JOIN
+                    (
+                          SELECT DISTINCT reactable_id
+                          FROM reactions
+                          WHERE reactable_type = 'Comment'
+                          AND created_at > (now() - interval '224' day)
+                          EXCEPT
+                          SELECT DISTINCT reactable_id
+                          FROM reactions
+                          WHERE reactable_type = 'Comment'
+                          AND created_at > (now() - interval '224' day)
+                          AND category IN ('thumbsdown', 'vomit')) AS negative_reactions
+                ON comments.id = negative_reactions.reactable_id
+                INNER JOIN
+                    (
+                          SELECT count(id) AS number_of_comments,
+                                 user_id AS comment_counts_user_id
+                          FROM comments
+                          WHERE created_at >= (now() - interval '7' day)
+                          GROUP BY user_id) AS comment_counts
+                          ON comments.user_id = comment_counts_user_id
+                          AND comment_counts.number_of_comments > 1
+                                /* Don’t select anything older than 224 days ago, or 32 weeks ago */
+                          WHERE created_at > (now() - interval '224' day)
+                          AND created_at < (now() - interval '3' day)
```

Comment:
The query now filters comments so they're at least 3 days old before they count towards the badge. This is intended so mods have time to flag comments before the badge is awarded to users [as per this conversation](https://github.com/forem/forem/issues/17310#issuecomment-1110045254)